### DISCUSSION TITLE
the error of Segmentation fault (core dumped)

### DISCUSSION THREAD

**Discussion by rtrtsdfsdf** [wfrest/wfrest#134] [Q&A] [answered]
PLEASE HELP ME!
I use the example of the following files, main.cc and  makefile .It is just the example.
It runs with error when I access http://127.0.0.1:8888/hello
[WFREST] POST   /post
[WFREST] GET    /multi
[WFREST] POST   /multi
[WFREST] GET    /hello
[WFREST] GET    /data
Segmentation fault (core dumped)

main.cc
`

      #include "workflow/WFFacilities.h"
      #include <csignal>
      #include "wfrest/HttpServer.h"
      
      using namespace wfrest;
      
      static WFFacilities::WaitGroup wait_group(1);
      
      void sig_handler(int signo)
      {
          wait_group.done();
      }
      
      int main()
      {
          signal(SIGINT, sig_handler);
      
          HttpServer svr;
      
          // curl -v http://ip:port/hello
          svr.GET("/hello", [](const HttpReq *req, HttpResp *resp)
          {
              resp->String("world\n");
          });
          // curl -v http://ip:port/data
          svr.GET("/data", [](const HttpReq *req, HttpResp *resp)
          {
              std::string str = "Hello world";
              resp->String(std::move(str));
          });
      
          svr.ROUTE("/multi", [](const HttpReq *req, HttpResp *resp)
          {
              std::string method(req->get_method());
              resp->String(std::move(method));
          }, {"GET", "POST"});
      
          // curl -v http://ip:port/post -d 'post hello world'
          svr.POST("/post", [](const HttpReq *req, HttpResp *resp)
          {
              // reference, no copy here
              std::string& body = req->body();
              fprintf(stderr, "post data : %s\n", body.c_str());
          });
      
          if (svr.track().start(8888) == 0)
          {
              svr.list_routes();
              wait_group.wait();
              svr.stop();
          } else
          {
              fprintf(stderr, "Cannot start server");
              exit(1);
          }
          return 0;
      }

`

makefile
`

              TARGET_OBJECT := all clean
              .PHONY: $TARGET_OBJECT
              LIB_WFREST = ~/my/c++/wfrest/_lib/
              LIB_WORKFLOW = ~/my/c++/workflow/_lib/
              INCLUDE_WFREST = ~/my/c++/wfrest/_include/
              INCLUDE_WORKFLOW = ~/my/c++/workflow/_include/
              INCLUDE := -I $(INCLUDE_WORKFLOW) -I $(INCLUDE_WFREST)
              LIB_DIR := -L $(LIB_WFREST) -L $(LIB_WORKFLOW)
              LIB := -l wfrest
              DBG_ARG := -g 
              all:
              #	@echo "INCLUDE = $(INCLUDE)"
	              g++  -Wall -fPIC -pipe -std=c++11 -fno-exceptions main.cc -o main.o $(LIB_DIR) $(LIB) $(INCLUDE) $(DBG_ARG) -lpthread
              
              clean:
	              rm -rf main.o
              

`

[main.zip](https://github.com/wfrest/wfrest/files/10120544/main.zip)

**Comment by chanchann** [comment]
I try this demo on my enviroment, it works well.

Can you supply more information about you platform?

**Reply by rtrtsdfsdf** [reply]
ubuntu 20.04
cmake version 3.16.3

CMake suite maintained and supported by Kitware (kitware.com/cmake).

what other information can I supply?

**Comment by chanchann** [comment]
<img width="1396" alt="image" src="https://user-images.githubusercontent.com/27816166/204734454-1b5e3206-e0f9-4c60-ac23-177a1338fa21.png">

**Reply by rtrtsdfsdf** [reply]
![image](https://user-images.githubusercontent.com/117705651/204734917-4b2b497e-5fc0-44b9-b6ee-cbe4fc2b3573.png)

![image](https://user-images.githubusercontent.com/117705651/204735016-a5d8495c-9e49-4177-afaf-b490f620ff93.png)

**Reply by rtrtsdfsdf** [reply]
![image](https://user-images.githubusercontent.com/117705651/204735410-42cd2c92-eede-4eed-9d5f-9a4f384f9099.png)

**Comment by rtrtsdfsdf** [comment]
Is there something I did wrong?
Why does this error occur?
Is it my version?

**Comment by chanchann** [comment]
<img width="1470" alt="image" src="https://user-images.githubusercontent.com/27816166/204736417-16c40f5d-6149-46b6-850a-835c32a2f5d5.png">

**Reply by chanchann** [reply]
I try it on ubuntu20, all is well.

**Reply by chanchann** [reply]
Which workflow library do you link?

**Reply by rtrtsdfsdf** [reply]
The latest version of workflow, I don't know how to check the workflow
version of，But git pull is up to date
![image](https://user-images.githubusercontent.com/117705651/204737187-eb1d3202-ddd7-4e05-8b11-43eb50bc01ca.png)

**Comment by chanchann** [comment]
Are you following the steps to link the correct workflow lib? 

```
git clone --recursive https://github.com/wfrest/wfrest
```

**Reply by rtrtsdfsdf** [reply]
![image](https://user-images.githubusercontent.com/117705651/204737881-e02cde26-1490-4338-bd19-b883ee7112b3.png)

**Comment by chanchann** [comment]
I suggest you trying to link wfrest's submodule workflow.

Following the steps I mentioned above.

**Reply by rtrtsdfsdf** [reply]
![image](https://user-images.githubusercontent.com/117705651/204738369-b8a6d2b6-05cf-4540-90ff-6f460d63dfe2.png)

**Comment by chanchann** [comment, marked as answer]
And you can try curl command first.

**Reply by rtrtsdfsdf** [reply]
I do this
git clone --recursive https://github.com/wfrest/wfrest
cd wfrest
make
sudo make install

and then I make the main.cpp, the error is same.
the curl is in same error.
![image](https://user-images.githubusercontent.com/117705651/204740572-1a4fdefc-1187-4919-b778-5af19385178d.png)
![image](https://user-images.githubusercontent.com/117705651/204740623-942dff69-bb60-4bc8-b199-32b9ff8aa03d.png)

**Reply by rtrtsdfsdf** [reply]
this is the g++version of mine.
![image](https://user-images.githubusercontent.com/117705651/204740907-f6f5788d-c18d-4af4-b96d-0e609c7e8f57.png)

**Reply by rtrtsdfsdf** [reply]
Thank you, I solved this problem, the workflow I used is the version on github, not wfrest/workflow,
There is an error in my workflow version, and the workflow under wfrest is normal, everything is OK
![image](https://user-images.githubusercontent.com/117705651/204742933-dcc2019f-6453-4532-b759-494f0269eaee.png)
![image](https://user-images.githubusercontent.com/117705651/204743082-dbd69769-c4c8-4fb7-8002-34a03c66846c.png)

**Comment by chanchann** [comment]
👏
